internal/pkg/cli/storage_init.go

```diff
@@ -184,9 +184,10 @@ type initStorageOpts struct {
 	initStorageVars
 	appName string
 
-	fs    afero.Fs
-	ws    wsAddonManager
-	store store
+	fs      afero.Fs
+	wsAddon wsAddonManager
+	wsWkld  wsWlReader
```

Comment:
nit what do you think of writing this as:
```suggestion
	ws interface {
	   wsAddonManager
	   wsWlReader
	}
```
like here: https://go.dev/play/p/Lw4aEhojPR_E

Reply:
Having anonymous interface does not work super pretty with go mock : (

However, this got me thinking a little. Indeed, having them as two dependency fields adds very little value to the abstraction. 

So I ended up regrouping the interfaces, relocating the addon functionality to workload/env interfaces instead of having an addon interface:
```
type wsEnvironmentReader interface {
	wsEnvironmentsLister
	EnvOverridesPath() string
	ReadEnvironmentManifest(mftDirName string) (workspace.EnvironmentManifest, error)
	EnvAddonFilePath(fName string) string // <----
}

type wsWorkloadReader interface {
	manifestReader
	WorkloadExists(name string) (bool, error)
	WorkloadAddonFilePath(wkldName, fName string) string // <----
}

type wsWorkloadReadWriter interface {
	wsWorkloadReader
	wsWriter
}

type wsReadWriter interface { // <----
	wsWorkloadReadWriter
	wsEnvironmentReader
}
```